### DISCUSSION TITLE
Pyright does not recognise Proto types

### DISCUSSION THREAD

**Discussion by superlopuh** [onnx/onnx#5865] [Q&A] [answered]
What's the recommended way to use the onnx API in a project that uses Pyright for type checking? It type checks all the imports to `Unknown`, meaning that there are type errors at every point of use.

I see that the original definition for all the Protos is deserialized, could this serialization be used for type stubs?

**Comment by justinchuby** [comment]
The pyi files were missing in version 1.15. It should be fixed in the next version. You may verify this by installing `onnx-weekly`

**Reply by superlopuh** [reply]
Thank you, do you know when that'll be released? Is it a weekly schedule?

**Reply by justinchuby** [reply, marked as answer]
onnx-weekly with the fix has been released already. You may uninstall onnx and pip install onnx-weekly instead. The next ONNX stable release should be in the coming weeks.
